gnd/src/main/java/com/google/android/gnd/repository/FeatureRepository.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.gnd.repository;
+
+import com.google.android.gnd.model.Mutation;
+import com.google.android.gnd.model.Project;
+import com.google.android.gnd.model.feature.Feature;
+import com.google.android.gnd.model.feature.FeatureMutation;
+import com.google.android.gnd.persistence.local.LocalDataStore;
+import com.google.android.gnd.persistence.remote.RemoteDataEvent;
+import com.google.android.gnd.persistence.remote.RemoteDataStore;
+import com.google.android.gnd.persistence.sync.DataSyncWorkManager;
+import com.google.common.collect.ImmutableSet;
+import io.reactivex.Completable;
+import io.reactivex.Flowable;
+import io.reactivex.Maybe;
+import java8.util.Optional;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+/**
+ * Coordinates persistence and retrieval of {@link Feature} instances from remote, local, and in
+ * memory data stores. For more details on this pattern and overall architecture, see
+ * https://developer.android.com/jetpack/docs/guide.
+ */
+@Singleton
+public class FeatureRepository {
+  private final LocalDataStore localDataStore;
+  private final ProjectRepository projectRepository;
+  private final DataSyncWorkManager dataSyncWorkManager;
+
+  @Inject
+  public FeatureRepository(
+      LocalDataStore localDataStore,
+      RemoteDataStore remoteDataStore,
+      ProjectRepository projectRepository,
+      DataSyncWorkManager dataSyncWorkManager) {
+    this.localDataStore = localDataStore;
+    this.projectRepository = projectRepository;
```

Comment:
I forget off the top of my head, but don't features and projects have foreign keys? If that's the case, don't we need to ensure that projectRepo.local/remoteDataStore and FeatureRepo.local/remoteDataStore are equivalent?

Reply:
UUIDs are being used as PKs and FKs, and we use the same ones for local and remote. How would we ensure they're equivalent? Not sure I follow..

Reply:
P.S. Trying to focus on splitting up the repo now without changing actual functionality for now. If relevant please create an issue with the suggestion and assign to me for follow-up. Thanks for all the input!